cmd/hypper/search.go

```diff
@@ -0,0 +1,71 @@
+/*
+Copyright The Helm Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"github.com/Masterminds/log-go"
+	"github.com/rancher-sandbox/hypper/pkg/search"
+	"github.com/spf13/cobra"
+)
+
+const searchDesc = `
+Search reads through all of the repositories configured on the system, and
+looks for matches. Search of these repositories uses the metadata stored on
+the system.
+
+It will display the latest stable versions of the charts found. If you
+specify the --devel flag, the output will include pre-release versions.
+If you want to search using a version constraint, use --version.
+
+Examples:
+
+    # Search for stable release versions matching the keyword "nginx"
+    $ hypper search nginx
+
+    # Search for release versions matching the keyword "nginx", including pre-release versions
+    $ hypper search nginx --devel
+
+    # Search for the latest stable release for nginx-ingress with a major version of 1
+    $ hypper search nginx-ingress --version ^1.0.0
+
+Repositories are managed with 'hypper repo' commands.
+`
+
+func newSearchCmd(logger log.Logger) *cobra.Command {
+	o := &search.RepoOptions{}
+
+	cmd := &cobra.Command{
+		Use:   "search [keyword]",
```

Comment:
I guess this is about preferences, but here we are implementing the new command as `hypper search [foo]`, instead of `hypper search repo [foo]`, which would be analogous to helm's `helm search repo [foo]`. Is that what we want? (just in case, as people may expect it on the latter).

Reply:
I could just modify it to be `repo search` in case we want to expand it further because I agree, coming from helm you would expect it to work like that

Reply:
I would be happy with the command being `hypper search repo`, just to mimic Helm's ui. No need to bring the `hypper search hub`, indeed. Who knows, maybe in the future we will have `hypper search cache` :).

Reply:
Now this seems to be addressed, by looking at the tests and newSearchRepoCmd(), but I think we missed the `const searchRepoDesc` examples.

Reply:
not as far as I can see, the same text is in place.

Reply:
shouldn't they be `$ hypper search repo nginx` instead of `$ hypper repo search nginx`?

Reply:
woops, dyslexia galore! Fixed!